### ISSUE TITLE
Add regex support for `colorNames`

### ISSUE THREAD

**Issue by fcisio** [atanasster/babel-plugin-theme-ui#1]
Hey @atanasster 
I'm stoked with this plugin, I think it works great!

One thing I think would be crucial to add is support for regex declarations in the `colorNames` object.

That way, we could set a pattern that would work everywhere. For example:
`[babelThemeUI, { colorNames: ["--bg-*", "--color-*"] }]`
or event better
`[babelThemeUI, { colorNames: ["--x-*"] }]`

Here's an input that would then work:
```js
a: {
  color: "primary",
  textDecoration: "none",
  "--bg-hover": "primary",
  "--bg-anything": "secondary",
  "--color": "primary",
  "--color-visited": "secondary",
},
// or
a: {
  color: "primary",
  textDecoration: "none",
  "--x-bg-hover": "primary",
  "--x-bg-anything": "secondary",
  "--x-text": "primary",
  "--x-text-visited": "secondary",
},
```

Also, in the example on this repo, I see that you've includes names that already work with Theme UI ('color', 'bg').
We could also make sure that only the strings starting with "--" will be transformed. That way, we ensure **only custom properties can be passed in.**

**Comment by atanasster** [comment]
Thanks for the feedback.

I will add a pattern for color names, great idea.

I am indeed adding the default color names by default - I thought this would make faster access since the colors will be resolved at build time. But I can also leave the default colorNames empty by default - what do you think?

**Comment by fcisio** [comment]
I would **only transform CSS Properties** with the plugin. So only register names starting with "--".

We don't want to mess with the native implementation, I think it's a breeding ground for a lot of unwanted side-effects. There are also many more properties that use the colors scale:

> color, background-color, border-color, caret-color, border-top-color, border-bottom-color, border-left-color, border-right-color, outline-color, fill, stroke

If the goal is to process those values at build time, then I think it qualifies for another plugin or a specific option.
Like "transformAll" that would transform all the color properties of Theme UI.

> With this, we could have all those names right in the codebase, making it easy to update it as new CSS properties appear and as Theme UI updates their specs. [See Theme UI Specs](https://theme-ui.com/theme-spec)

#### Also this should not be an issue when working with SSR or SSG like Gatsby or Next

👍

**Comment by atanasster** [comment]
@fcisio - great suggestions

I released a new version 0.5.0 -

-- the colorNames are now by default empty, however using the `transformNativeColors` option will add the native theme-ui color scales

-- colorNames can now use pattern matching using [micromatch](https://github.com/micromatch/micromatch)

Closing this for now
